File: RepairShop/RepairShop/Models/Conversation.swift
//
//  Conversation.swift
//  RepairShop
//
//  Created by Razvan Dumitriu on 21.05.2023.
//

import Foundation

struct Conversation {
    let userId: String
    let userName: String
    let repairshopId: String
    let repairshopName: String
    let messages: [Message]
    
    struct Message {
        let userId: String
        let message: String
        let date: Date
    }
}

File: backendmds/model/conversation.js
const mongoose = require("mongoose");

const conversationSchema = new mongoose.Schema({
    user_id: { type: String, default: null },
    repairshop_id: { type: String, default: null },
    messages: [{
        message: { type: String, default: null },
        date: { type: Date, default: null }
    }]
});

module.exports = mongoose.model("conversation", conversationSchema);